zcash_client_sqlite/src/wallet/init/migrations/shardtree_support.rs

```diff
@@ -0,0 +1,194 @@
+        commitment_tree::SqliteShardStore,
+        init::{migrations::received_notes_nullable_nf, WalletMigrationError},
+    },
+    SAPLING_TABLES_PREFIX,
+};
+
+pub(super) const MIGRATION_ID: Uuid = Uuid::from_fields(
+    0x7da6489d,
+    0xe835,
+    0x4657,
+    b"\x8b\xe5\xf5\x12\xbc\xce\x6c\xbf",
+);
+
+pub(super) struct Migration;
+
+impl schemer::Migration for Migration {
+    fn id(&self) -> Uuid {
+        MIGRATION_ID
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        [received_notes_nullable_nf::MIGRATION_ID]
+            .into_iter()
+            .collect()
+    }
+
+    fn description(&self) -> &'static str {
+        "Add support for receiving storage of note commitment tree data using the `shardtree` crate."
+    }
+}
+
+impl RusqliteMigration for Migration {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &rusqlite::Transaction) -> Result<(), WalletMigrationError> {
+        // Add commitment tree sizes to block metadata.
+        transaction.execute_batch(
+            "ALTER TABLE blocks ADD COLUMN sapling_commitment_tree_size INTEGER;
+             ALTER TABLE blocks ADD COLUMN orchard_commitment_tree_size INTEGER;
+             ALTER TABLE sapling_received_notes ADD COLUMN commitment_tree_position INTEGER;",
+        )?;
+
+        // Add shard persistence
+        transaction.execute_batch(
+            "CREATE TABLE sapling_tree_shards (
+                shard_index INTEGER PRIMARY KEY,
+                subtree_end_height INTEGER,
+                root_hash BLOB,
+                shard_data BLOB,
+                contains_marked INTEGER,
+                CONSTRAINT root_unique UNIQUE (root_hash)
+            );
+            CREATE TABLE sapling_tree_cap (
+                -- cap_id exists only to be able to take advantage of `ON CONFLICT`
+                -- upsert functionality; the table will only ever contain one row
+                cap_id INTEGER PRIMARY KEY,
+                cap_data BLOB NOT NULL
+            );",
+        )?;
+
+        // Add checkpoint persistence
+        transaction.execute_batch(
+            "CREATE TABLE sapling_tree_checkpoints (
+                checkpoint_id INTEGER PRIMARY KEY,
+                position INTEGER
+            );
+            CREATE TABLE sapling_tree_checkpoint_marks_removed (
+                checkpoint_id INTEGER NOT NULL,
+                mark_removed_position INTEGER NOT NULL,
+                FOREIGN KEY (checkpoint_id) REFERENCES sapling_tree_checkpoints(checkpoint_id)
+                ON DELETE CASCADE
+            );",
+        )?;
+
+        let shard_store =
+            SqliteShardStore::<_, sapling::Node, SAPLING_SHARD_HEIGHT>::from_connection(
+                transaction,
+                SAPLING_TABLES_PREFIX,
+            )?;
+        let mut shard_tree: ShardTree<
+            _,
+            { sapling::NOTE_COMMITMENT_TREE_DEPTH },
+            SAPLING_SHARD_HEIGHT,
+        > = ShardTree::new(shard_store, 100);
```

Comment:
Use `PRUNING_HEIGHT` constant.

Reply:
Also, self-nit, `PRUNING_HEIGHT` isn't a height, it's a depth and should be renamed accordingly.